packages/analytics-js-integrations/src/integrations/GA4/browser.js

```diff
@@ -39,6 +39,7 @@ export default class GA4 {
     this.overrideClientAndSessionId = config.overrideClientAndSessionId || false;
     this.sdkBaseUrl =
       removeTrailingSlashes(config.sdkBaseUrl) || 'https://www.googletagmanager.com';
+    this.isExtendedGa4_V2 = config.isExtendedGa4_V2 || false;
```

Comment:
We're polluting GA4 implementation with the "v2" context. 
Ideally, the common code parts should be extracted into a separate module and version-specific logic should remain in individual integration code bases.

It does not need to be addressed now. So, please create a refactoring ticket and resolve it yourself.

Reply:
Agree, need to bring loadScript outside for common code & have 2 different implementation for GA4 & GA4_v2

Reply:
The problem is the load script updates member variables of the parent class, just bringing load script wont work we need to figure out an alternative

Reply:
@utsabc I see you tried to avoid code duplication but that should be fine instead of making it complex. `getClientDetails` method could also have been avoided in this class with little code duplication

Reply:
Yeah I will create backlog